scio-test/src/main/scala/com/spotify/scio/testing/JobTest.scala

```diff
@@ -65,26 +65,32 @@ import scala.util.control.NonFatal
  */
 object JobTest {
 
-  case class Builder(className: String, cmdlineArgs: Array[String],
+  private case class BuilderState(className: String, cmdlineArgs: Array[String],
                      inputs: Map[TestIO[_], Iterable[_]],
                      outputs: Map[TestIO[_], SCollection[_] => Unit],
-                     distCaches: Map[DistCacheIO[_], _]) {
+                     distCaches: Map[DistCacheIO[_], _], wasRunInvoked: Boolean = false)
```

Comment:
So the way Builder worked here was to create new instances of Builder, thus in the `EngineerProofJobTest` we would have access only to the initial Builder instances not the one that `run` is invoked on. I don't really like this solution much, essentially what I need is `pointer` or `container` that will store the eventual state of the builder.